index.js

```diff
@@ -435,15 +435,18 @@ export default class OneSignal {
         return RNOneSignal.userProvidedPrivacyConsent();
     }
 
-    static setExternalUserId(externalId, callback) {
+    static setExternalUserId(externalId, callback=function(){}) {
         if (!checkIfInitialized()) return;
 
-        if (callback === undefined)
-            callback = function(){};
-
         RNOneSignal.setExternalUserId(externalId, callback);
     }
 
+    static setExternalUserId(externalId, authHashToken, callback=function(){}) {
```

Comment:
Function overloading doesn't work like this in JS

Reply:
Function defaulting is an ES6 feature however I believe we still need to support ES5 for now.
We should keep this code to default.
```javascript
if (callback === undefined)
   callback = function(){};
```

Reply:
We need to use `typeof` to see what the 2nd param really is so we don't break anyone's existing usage of `setExternalUserId`.

Reply:
updated